src/components/views/right_panel/UserInfo.tsx

```diff
@@ -187,9 +187,15 @@ function DeviceItem({userId, device}: {userId: string, device: IDevice}) {
         verifyDevice(cli.getUser(userId), device);
     };
 
-    const deviceName = device.ambiguous ?
-        (device.getDisplayName() ? device.getDisplayName() : "") + " (" + device.deviceId + ")" :
-        device.getDisplayName();
+    let deviceName;
+    if (device.getDisplayName() === null || device.getDisplayName().trim() === "") {
+        deviceName = device.deviceId;
+    } else {
+        deviceName = device.ambiguous ?
+            device.getDisplayName() + " (" + device.deviceId + ")" :
+            device.getDisplayName();
+    }
```

Comment:
We can use a bit of TS here
Something like this should work

```suggestion
    if (!device.getDisplayName()?.trim()) {
        deviceName = device.deviceId;
    } else {
        deviceName = device.ambiguous ?
            device.getDisplayName() + " (" + device.deviceId + ")" :
            device.getDisplayName();
    }
```

But wouldn't a better way to do this be setting `device.ambiguous` to True whenever there's no displayName, that's certainly ambiguous IMO

Reply:
Yes it's ambiguous. Doing it that way would just put the device ID in parentheses for no reason and I thought it looked kind of odd.